src/actions/points-system/calculate-user-points.ts

```diff
@@ -0,0 +1,69 @@
+import { Op } from "sequelize";
+import db, { sequelizeConnection } from "src/db";
+import logger from "src/utils/logger-handler";
+
+export const name = "CalculatePointsDaily";
+export const schedule = "0 0 * * *";
+export const description = "Calculate total points and update users' total points";
+export const author = "Vitor Hugo";
+
+export async function action() {
+  logger.info(`${name} start`);
+
+  const events = await db.points_events.findAll({
+    where: {
+      pointsCounted: false,
+    },
+    order: ["userId"],
+    raw: true
+  });
+
+  if (!events?.length) {
+    logger.info(`${name} no counted events not found`);
+    return;
+  }
+
+  const pointsByUser = {} as { [userId: string]: { userId: number, total: number, eventsIds: number[] } };
+
+  for (const event of events) {
```

Comment:
We don't need the `transaction` nor do we need to update users on each iteration, we can simply use `bulkUpdate` for the users and then simply target the parsed event ids via `[Op.in]`
```ts
const points = {};

for (const event of events) {
  if (!points[event.userId])
    points[event.userId] = {eventId: [], points: 0};
  points[event.userId].points += event.pointsWon;
  points[event.userId].eventId.push(event.id);
}

const updateUserPoints = Object.entries(points).map(([id, {points}]) => ({id, points}))
const parsedEventIds = Object.values(points).map(({eventId}) => eventId).flat()

await db.users.bulkUpdate(
  updateUserPoints,
  {fields: "pointsWon", where: {id: updateUserPoints.map(({id}) => id)}
)

await db.points_events.update({pointsCounted: true}, {where: {id: [Op.in]: parsedEventIds}})

```